### PR TITLE
10.0 pos order cancel add option free reason

### PR DESCRIPTION
**Pull request by gaelTorrecillas**
hello @yelizariev

I add an option to choose if we accept a free reason ( normal mode) or if the salesman must choose one or more specific reasons.

In the second point, we desable the textarea ( input box) and the button "cancel". The button "validate" is disable when the salesman don't choose the reason.

### CODE REVIEW

**Comment by flotho** [comment]
Hi @yelizariev ,

Could you help us to understand what's going wrong on this PR : https://travis-ci.org/it-projects-llc/pos-addons/jobs/280507895
We don't see flake8 error.
Regards

File: pos_order_cancel/models/models.py [hunk 1]
```diff
@@ -135,3 +135,8 @@ def create(self, values):
             canceled_date = fields.Datetime.to_string(canceled_date)
             values['canceled_date'] = canceled_date
         return super(PosOrderLineCanceled, self).create(values)
+    
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+    
+    accept_free_reason=fields.Boolean(string="accept free reason",help="the field to send a reason is activate",default=True)
```
Comment (yelizariev) [inline pos_order_cancel/models/models.py]: Can it be renamed to something like ``allow_custom_reason``, because ``freee`` has different meanings...

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Comment by yelizariev** [comment]
It has reference to the code with explanation of lint warning.
```
pos_multi_session/tests/test_test.py:4:1: F401 'odoo.fields' imported but unused
pos_multi_session/tests/test_test.py:5:1: F401 'odoo.tools.mute_logger' imported but unused
pos_multi_session/tests/test_test.py:5:1: F401 'odoo.tools.float_compare' imported but unused
pos_multi_session/tests/test_test.py:9:1: E302 expected 2 blank lines, found 1
pos_order_cancel/models/models.py:138:1: W293 blank line contains whitespace
pos_order_cancel/models/models.py:139:1: E302 expected 2 blank lines, found 1
pos_order_cancel/models/models.py:141:1: W293 blank line contains whitespace
pos_order_cancel/models/models.py:142:23: E225 missing whitespace around operator
pos_order_cancel/models/models.py:142:66: E231 missing whitespace after ','
pos_order_cancel/models/models.py:142:112: E231 missing whitespace after ','
```

**Review by yelizariev — APPROVED** [review]

**Comment by yelizariev** [comment]
@gaelTorrecillas Bon travail!

@GabbasovDinar or @ilmir-k please review before I will merge

**Comment by ilmir-k** [comment]
@gaelTorrecillas 

We have some bug when "Allow custom cancellation reason" disabled. We see 'undefined' in backend near a reason specified after cancellation. It happens for both a line and order cancellations.
test

**Comment by gaelTorrecillas** [comment]
hi @ilmir-k ,

I fix the problem. thank you for your comment.

**Review by ilmir-k — APPROVED** [review]
approuvé :+1:

**Comment by gaelTorrecillas** [comment]
@ilmir-k ,

could you explain me what I see 
`Merge branch '10.0' into 10.0-pos_order_cancel-add-option-free-reason
01cb85d `
in the log but the PR is always open (so no merge)

ps : sorry french level on english

**Comment by ilmir-k** [comment]
@gaelTorrecillas 

yes, it is because we update your branch periodically, due to it doesn't contain the latest commits we merged. Generally speaking we rebase your branch with the base branch. 

E.g., let's say you pushed the updates on branch 10.0, but at this time another person made some updates (commits), which have been merged before your one. So as a result your branch have not these updates and we have to merge the changes from branch 10.0 to your branch. Only then we are able to merge your PR to base branch. Hope it is clear now.

**Comment by gaelTorrecillas** [comment]
hello @ilmir-k , @yelizariev ,

I rebase my branch with your branch 10.0.

**Comment by ilmir-k** [comment]
@gaelTorrecillas 

It seems that you rebase your branch incorrectly.
Please be ensure that you made the following steps:
If your remotes like this: 
```
origin     https://github.com/gaelTorrecillas/pos-addons.git
upstream    https://github.com/it-projects-llc/pos-addons.git
```
you need execute:
```
git fetch upstream 10.0
git rebase upstream/10.0
git push --force-with-lease origin 10.0-pos_order_cancel-add-option-free-reason
```

Please check it

**Comment by gaelTorrecillas** [comment]
@ilmir-k ,

I rebase my branch, and I suppose that this branch is clean. thank for your help.

**Comment by gaelTorrecillas** [comment]
Hello,

@ilmir-k , @yelizariev  what do you think I did not add any code but I changed branch because the commit list was very chaotic. Do I want to rebase my branch because your branch 10.0 is ahead?

**Comment by yelizariev** [comment]
@gaelTorrecillas there is one more requirement now to merge this PR as green one:
you need to have phantom_js tests.

You can copy-paste this one: https://github.com/it-projects-llc/pos-addons/tree/10.0/pos_disable_payment/tests
it just checks that POS is loaded without error. 
But you are welcome to extend it to make further tests (add item, remove item, select reason). You can find some information here: http://odoo-development.yelizariev.dev.it-projects.info/dev/tests/js.html

**Comment by yelizariev** [comment]
Use this link : https://github.com/it-projects-llc/pos-addons/commit/6f857b8ba16c13690c39586cef77ec1d36542a12
it shows all minimal updates to activate the test

**Comment by yelizariev** [comment]
@gaelTorrecillas you need to include js and other files as well. See https://github.com/it-projects-llc/pos-addons/commit/6f857b8ba16c13690c39586cef77ec1d36542a12

**Comment by gaelTorrecillas** [comment]
@yelizariev ,

I fix it, but coveralls is not good

**Comment by yelizariev** [comment]
@gaelTorrecillas coverage points to others modules (I don't know why). So, your updates are actually green.

**Comment by yelizariev** [comment]
:+1:

**Comment by flotho** [comment]
Thanks @yelizariev, 

It looks like a great contribution from @gaelTorrecillas , don't you agree?

**Pull request merged**